src/tls/keypair.h

```diff
@@ -17,67 +17,159 @@
 #include <mbedtls/eddsa.h>
 #include <memory>
 
-#if CURVE_CHOICE_SECP384R1
+namespace tls
+{
+  enum class CurveImpl
+  {
+    secp384r1 = 1,
+    curve25519 = 2,
+    secp256k1_mbedtls = 3,
+    secp256k1_bitcoin = 4,
+
+#if LEDGER_CURVE_CHOICE_SECP384R1
+    ledger_curve_choice = secp384r1,
```

Comment:
There's something that I'm probably missing but I'm not sure the curve choice should mention "ledger" as it is used in a bunch of places that are not directly related to the ledger itself (e.g. node and network identity).